components/service/glean/src/main/java/mozilla/components/service/glean/private/TimespanMetricType.kt

```diff
@@ -29,67 +31,85 @@ data class TimespanMetricType(
 
     private val logger = Logger("glean/TimespanMetricType")
 
+    // The identifier for the timer managed by this metric.
+    private var timerId: GleanTimerId? = null
+
     /**
-     * Start tracking time for the provided metric and associated object. This
-     * records an error if it’s already tracking time (i.e. start was already
-     * called with no corresponding [stopAndSum]): in that case the original
+     * Start tracking time for the provided metric.
+     * This records an error if it’s already tracking time (i.e. start was already
+     * called with no corresponding [stop]): in that case the original
      * start time will be preserved.
-     *
-     * @param timerId The object to associate with this timing.  This allows
-     * for concurrent timing of events associated with different objects to the
-     * same timespan metric.
      */
-    fun start(timerId: Any) {
+    fun start() {
         if (!shouldRecord(logger)) {
             return
         }
 
-        TimingManager.start(this, timerId)
+        if (timerId != null) {
+            ErrorRecording.recordError(
+                    this,
+                    ErrorRecording.ErrorType.InvalidValue,
+                    "Timespan already started",
+                    logger
+            )
+            return
+        }
+
+        timerId = TimingManager.start(this)
     }
 
     /**
-     * Stop tracking time for the provided metric and associated object. Add the
-     * elapsed time to the time currently stored in the metric. This will record
+     * Stop tracking time for the provided metric.
+     * Add the elapsed time to the time currently stored in the metric. This will record
      * an error if no [start] was called.
-     *
-     * @param timerId The object to associate with this timing.  This allows
-     * for concurrent timing of events associated with different objects to the
-     * same timespan metric.
      */
-    fun stopAndSum(timerId: Any) {
+    fun stop() {
         if (!shouldRecord(logger)) {
             return
         }
 
-        TimingManager.stop(this, timerId)?.let { elapsedNanos ->
-            @Suppress("EXPERIMENTAL_API_USAGE")
-            Dispatchers.API.launch {
-                TimespansStorageEngine.sum(this@TimespanMetricType, timeUnit, elapsedNanos)
+        if (timerId == null) {
```

Comment:
nit, you could be more Kotlin idomatic if this was a `?:` attached to the `timerId?.let { ...` below.

Something like this maybe:
```
timerId?.let { id ->
    ...
} ?: ErrorRecording.recordError(
    ...
)
```

Reply:
You got me here! Great catch!

Reply:
Errata: it appears that I'm not able to do that, due to the return type of the two blocks being different.